lms/djangoapps/courseware/model_data.py

```diff
@@ -875,6 +877,8 @@ def delete(self, key):
 
         Raises: KeyError if key isn't found in the cache
         """
+        if self.read_only:
```

Comment:
I assume this was added for completeness-sake, but not really needed for the crawler-case since, theoretically there shouldn't be any deletions?